private_transactions/src/lib.rs

```diff
@@ -0,0 +1,782 @@
+
+use_contract!(private, "PrivateContract", "res/private.json");
+
+/// Initialization vector length.
+const INIT_VEC_LEN: usize = 16;
+
+struct TransactionDetailsProvider<'a> {
+	client: &'a MiningBlockChainClient,
+}
+
+impl<'a> TransactionDetailsProvider<'a> {
+	pub fn new(client: &'a MiningBlockChainClient) -> Self {
+		TransactionDetailsProvider {
+			client: client,
+		}
+	}
+}
+
+impl<'a> TransactionQueueDetailsProvider for TransactionDetailsProvider<'a> {
+	fn fetch_account(&self, address: &Address) -> AccountDetails {
+		AccountDetails {
+			nonce: self.client.latest_nonce(address),
+			balance: self.client.latest_balance(address),
+		}
+	}
+
+	fn estimate_gas_required(&self, tx: &SignedTransaction) -> U256 {
+		tx.gas_required(&self.client.latest_schedule()).into()
+	}
+
+	fn is_service_transaction_acceptable(&self, _tx: &SignedTransaction) -> Result<bool, String> {
+		Ok(false)
+	}
+}
+
+/// Configurtion for private transaction provider
+#[derive(Default, PartialEq, Debug, Clone)]
+pub struct ProviderConfig {
+	/// Accounts that can be used for validation
+	pub validator_accounts: Vec<Address>,
+	/// Account used for signing public transactions created from private transactions
+	pub signer_account: Option<Address>,
+	/// Passwords used to unlock accounts
+	pub passwords: Vec<String>,
+}
+
+#[derive(Debug)]
+/// Private transaction execution receipt.
+pub struct Receipt {
+	/// Private transaction hash.
+	pub hash: H256,
+	/// Created contract address if any.
+	pub contract_address: Option<Address>,
+	/// Execution status.
+	pub status_code: u8,
+}
+
+/// Manager of private transactions
+pub struct Provider {
+	encryptor: Arc<Encryptor>,
+	config: ProviderConfig,
+	notify: RwLock<Vec<Weak<ChainNotify>>>,
+	transactions_for_signing: Mutex<SigningStore>,
+	transactions_for_verification: Mutex<VerificationStore>,
+	client: Arc<Client>,
+	accounts: Arc<AccountProvider>,
+}
+
+#[derive(Debug)]
+struct PrivateExecutionResult<T, V> where T: Tracer, V: VMTracer {
+	code: Option<Bytes>,
+	state: Bytes,
+	result: Executed<T::Output, V::Output>,
+}
+
+impl Provider where {
+	/// Create a new provider.
+	pub fn new(client: Arc<Client>, accounts: Arc<AccountProvider>, encryptor: Arc<Encryptor>, config: ProviderConfig) -> Result<Self, PrivateTransactionError> {
+		Ok(Provider {
+			config: config,
+			notify: RwLock::new(Vec::new()),
+			transactions_for_signing: Mutex::new(SigningStore::new()),
+			transactions_for_verification: Mutex::new(VerificationStore::new()),
+			client: client,
+			accounts: accounts,
+			encryptor: encryptor,
```

Comment:
recent version of rust has nice syntax sugar, so you can write:

```rust
Provider {
  config,
  client,
  accounts,
  encryptor,
  ..Default::default()
}
```

and it will initialize fields in exactly the same way